src/main/scala/chisel3/util/SRAM.scala

```diff
@@ -78,6 +81,45 @@ class MemoryReadWritePort[T <: Data](tpe: T, addrWidth: Int, masked: Boolean) ex
   }
 }
 
+@instantiable
+/** Object Module for SRAM, used for storing metadata of an SRAM. */
+class SRAMOM extends Class {
```

Comment:
The term "Object Model" is a rocket-chip term, not a Chisel one, and I'm not keen to introduce it here. I think we should stick to something more generic like "Metadata" or "Properties", any thoughts @seldridge or @mikeurbach?

Reply:
The Property term is what we're using in Chisel 🤷 The Property docs have some nuggets that we could also use: "non-hardware information", and "domain-specific data models" show up in the introductory paragraphs.

Reply:
I have been thinking about this a bit in out of tree projects as well, and thinking about what it would look like if certain Modules were expected to have Objects of certain Classes attached.

I personally don't like the term "metadata", since in some teams/flows, this information can be as important as the RTL. But I do think it would be good to pick a name that something we can use to refer to these Objects we always attach to certain Modules. I'm not sure what a good name would be, but I'll use "Description" as an example since I've seen it downstream. So this could be the SRAMDescription, but we could also have other Models like a hypothetical ALUDescription or whatever.

Reply:
I will also leave this as food for thought, which was something we discussed quite a bit while working on the Object Model dialect in CIRCT. Yes, we did pick, that name, and yes, I'm agreeing with Jack that that's maybe not the name to use here.

> Your domain model is not your object model. Your domain model is not your abstract data type model. Your object model is not your abstract data type model.

In my mind, Properties and Objects are the building blocks for representing an object model. But when you say an SRAM has some "description"/"metadata"/"whatever", that is about your domain model, and the object model is what you project your domain model into.

https://www.sicpers.info/2018/03/why-inheritance-never-made-any-sense/

Reply:
> In my mind, Properties and Objects are the building blocks for representing an object model. But when you say an SRAM has some "description"/"metadata"/"whatever", that is about your domain model, and the object model is what you project your domain model into.

This does make sense to me!

I'll change it to `SRAMDescription` in the following commit.